src/gt4py/next/program_processors/runners/dace/transformations/move_dataflow_into_if_body.py

```diff
@@ -456,11 +457,13 @@ def _update_symbol_mapping(
         symbols = if_block.sdfg.symbols
         for sym, symval in if_block.symbol_mapping.items():
             if sym not in symbols:
-                if_block.sdfg.add_symbol(
-                    sym,
-                    dace.codegen.tools.type_inference.infer_expr_type(symval, if_block.sdfg.symbols)
-                    or dace_dtypes.typeclass(int),
-                )
+                if sym == str(symval):
+                    sym_type = parent.symbols.get(sym, dace_dtypes.typeclass(int))
+                else:
+                    sym_type = dace.codegen.tools.type_inference.infer_expr_type(
+                        symval, if_block.sdfg.symbols
```

Comment:
```suggestion
                    sym_type = dace.codegen.tools.type_inference.infer_expr_type(
                        symval, parent.symbols
```

I am actually not sure, but I would say that we should use the parent SDFG here, because the parent sdfg defines them.
It might even be an error in DaCe, I will ask in the channel.

Reply:
I think this is not possible, because this would imply a certain order in which you can do something.
This the left hand side (key) of the mapping always refers to the symbols inside the nested SDFG and the right hand side (value) always refers to the symbols in the parent scope.
I made the same error when I reviewed the "scope PR".